File: homeworks/HW2/src/main/java/ru.atom/model/Tile.java
package ru.atom.model;

import ru.atom.geometry.Point;

public class Tile implements Positionable {
    private final int id;

    // Entity position on map grid
    private final Point position;

    private final Materials material;

    public enum Materials {
        GRASS, WALL, WOOD
    }

    public Tile(Point position, Materials material) {
        this.id = GameSession.getNextId();
        this.position = position;
        this.material = material;
    }

    @Override
    public int getId() {
        return id;
    }

    @Override
    public Point getPosition() {
        return position;
    }
}

File: bomberman/frontend/webapp/js/Tile.js
Tile = Entity.extend({
    /**
     * Entity position on map grid
     */
    position: {},

    /**
     * Bitmap dimensions
     */
    size: {
        w: 32,
        h: 32
    },

    /**
     * Bitmap animation
     */
    bmp: null,

    material: '',

    init: function(id, material, position) {
        this.id = id;
        this.material = material;
        this.position = position;
        var img;
        if (material == 'grass') {
            img = gGameEngine.tilesImgs.grass;
        } else if (material === 'Wall') {
            img = gGameEngine.tilesImgs.wall;
        } else if (material === 'Wood') {
            img = gGameEngine.tilesImgs.wood;
        }
        this.bmp = new createjs.Bitmap(img);

        this.bmp.x = position.x;
        this.bmp.y = position.y;

        gGameEngine.stage.addChild(this.bmp);
    },

    update: function() {
    },

    remove: function() {
        gGameEngine.stage.removeChild(this.bmp);
    }
});